tests/cloud/conftest.py

```diff
@@ -0,0 +1,170 @@
+        return str(os.path.join(self.onnx_dir_path(), self.model_name.replace("/", "_") + "_kv_clipped_fp16.onnx"))
+    
+    def model_hf_path(self):
+        return str(os.path.join(self.cache_dir,self.model_name))
+    
+    def base_path_and_generated_onnx_path(self):
+        return str(self.onnx_dir_path()), str(os.path.join(self.onnx_dir_path(), self.model_name.replace("/", "_") + "_kv_clipped_fp16.onnx"))
+    
+    def specialization_json_path(self):
+        return str(os.path.join(self.model_card_dir(), self.qpc_base_dir_name(), "specializations.json"))
+    
+    def custom_io_file_path(self):
+        if self.mxint8:
+            return str(os.path.join(self.onnx_dir_path(), "custom_io_int8.yaml"))
+        else:
+            return str(os.path.join(self.onnx_dir_path(), "custom_io_fp16.yaml"))
+
+@pytest.fixture
+def setup(model_name,num_cores,prompt,prompts_txt_file_path,aic_enable_depth_first,mos,cache_dir,hf_token,batch_size,prompt_len,ctx_len,mxfp6,mxint8,device_group):
+    """
+    It is a fixture or shared object of all testing script within or inner folder,
+    Args are coming from the dynamically generated tests method i.e, pytest_generate_tests via testing script or method
+    --------
+    Args: same as set up initialization
+    Return: model_setup class object
+    """
+    if hf_token == "NA" and prompts_txt_file_path == "NA":
+        return model_setup(model_name,num_cores,prompt,None,bool(aic_enable_depth_first),mos,cache_dir,None,batch_size,prompt_len,ctx_len,bool(mxfp6),bool(mxint8),device_group)
+    
+    elif prompts_txt_file_path == "NA":
+        return model_setup(model_name,num_cores,prompt,None,bool(aic_enable_depth_first),mos,cache_dir,hf_token,batch_size,prompt_len,ctx_len,bool(mxfp6),bool(mxint8),device_group)
+
+    elif hf_token == "NA":
+        return model_setup(model_name,num_cores,prompt,prompts_txt_file_path,bool(aic_enable_depth_first),mos,cache_dir,None,batch_size,prompt_len,ctx_len,bool(mxfp6),bool(mxint8),device_group)
+
+    else:
+        return model_setup(model_name,num_cores,prompt,prompts_txt_file_path,bool(aic_enable_depth_first),mos,cache_dir,hf_token,batch_size,prompt_len,ctx_len,bool(mxfp6),bool(mxint8),device_group)
+
+def pytest_generate_tests(metafunc):  
+    """
+    pytest_generate_tests hook is used to create our own input parametrization,
+    It generates all the test cases of different combination of input parameters which are read from the json file,
+    and passed to each testing script module.
+    -----------
+    Ref: https://docs.pytest.org/en/7.3.x/how-to/parametrize.html
+    """
+    json_data = None
+    json_file  = os.path.join(ROOT_DIR,"tests","HL_testing_input.json")
+    with open(json_file,'r') as file:
+        json_data =  json.load(file)
+    print("\n**************JSON data***************\n\n",json_data)
+
+    if ('model_name' in metafunc.fixturenames and 'num_cores' in metafunc.fixturenames and 'prompt' in metafunc.fixturenames

```

Comment:
Is this if condition required?

Reply:
Yes, required just because JSON does not support **NONE** keywords.

Reply:
Removed.

Reply:
Is this still required?